src/main/java/io/debezium/connector/vitess/VitessConnector.java

```diff
@@ -142,17 +143,22 @@ public List<Map<String, String>> taskConfigs(int maxTasks, List<String> currentS
                         prevGtidsPerShard.keySet(), currentShards);
             }
             final String keyspace = connectorConfig.getKeyspace();
+            // Check the configs in case there is a user specified GTID override
+            verifyShardGtidConfig();
+            Map<String, String> gtidsPerShard = getGtidsPerShardFromConfig();
```

Comment:
This logic is not correct, the shard/grids will need to come from the Kafka storage instead of from static config on the normal run.

The static config value is not used for the first run when there is no history in Kafka storage.

Reply:
Please don't merge in this PR yet, I will work with Tom separately to sort out some of the issues.

Reply:
Updated with this feedback, thanks!